### PR TITLE
TS 2

### PR DESCRIPTION
**Pull request by Kamahl19**
#312

### CODE REVIEW

File: src/app/DemoScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const mapStateToProps = (state: AppState) => ({
+  profile: selectProfile(state),
+});
+
+type Props = ReturnType<typeof mapStateToProps>;
+
+const DemoScreen = IsLoggedIn(({ profile }: Props) => (
+  <div>
+    ID: {profile!.id}
```
Comment (MiroslavPetrik) [inline src/app/DemoScreen.tsx]: toto neni cool, skor if profile ? null : screen

to by sme mali ! vsade, lebo kazda featura zacne s empty stavom, sice budu prazdne struktury ako array alebo obj, ale aj null moze byt

File: src/app/store/index.ts [hunk 2]
```diff
@@ -1,6 +1,12 @@
 import createRootReducer from './rootReducer';
 
+/**
+ * Re-export from store
+ */
 export { store } from './configureStore';
 export { default as StorePersistGate } from './StorePersistGate';
 
+/**
+ * AppState - type representing root state-tree
```
Comment (MiroslavPetrik) [inline src/app/store/index.ts]: maybe name it RootState?

File: src/bootstrap.ts [hunk 3]
```diff
@@ -22,4 +22,4 @@ import 'app/store';
 /**
  * All actions types for typesafe-actions
  */
-import './actions';
+import 'actions';
```
Comment (MiroslavPetrik) [inline src/bootstrap.ts]: ako toto funguje? aj ked je `app/something`

File: src/common/services/user/index.ts [hunk 4]
```diff
@@ -83,7 +83,7 @@ export default combineReducers({
 /**
  * SELECTORS
  */
-export const selectUser = (state: AppState) => state.user;
+export const selectUser = (state: AppState): UserState => state.user;
```
Comment (MiroslavPetrik) [inline src/common/services/user/index.ts]: hmm? to naco

File: src/app/DemoScreen.tsx (hunk 1 above)
Comment (Kamahl19) [reply src/app/DemoScreen.tsx]: jasne ze je to zle, to je len throwaway co najsimple component

File: src/bootstrap.ts (hunk 3 above)
Comment (Kamahl19) [reply src/bootstrap.ts]: https://github.com/facebook/create-react-app/pull/6656

File: src/common/components/ErrorBoundary/index.tsx [hunk 5]
```diff
@@ -5,15 +5,15 @@ type Props = {
...
+  readonly hasError: boolean;
 };
 
 export default class ErrorBoundary extends Component<Props, State> {
   static getDerivedStateFromError() {
     return { hasError: true };
   }
 
-  state = {
+  readonly state = {
```
Comment (MiroslavPetrik) [inline src/common/components/ErrorBoundary/index.tsx]: co toto?

    this.state.hasError = false;
v renderi vobec nehlasi chybu

ak spravim
state: State = { ... }

tak hlasi

Comment (MiroslavPetrik) [reply src/common/components/ErrorBoundary/index.tsx]: aha nieje to deep.

`this.state = {foo: 'bar'}` uz hlasi chybu

Comment (MiroslavPetrik) [reply src/common/components/ErrorBoundary/index.tsx]:
```
 this.state = {hasError: true};
```
hlasi chybu len ak mam obe

```
  readonly state: State = {
    hasError: false,
  };
```

Comment (MiroslavPetrik) [reply src/common/components/ErrorBoundary/index.tsx]: WE WANT BOTH STATE INSTANCE PROPERTY AND IT'S KEYS TO BE READONLY.

FIX IT PLEASE.

File: tsconfig.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "allowJs": false,
+    "allowJs": true,
```
Comment (MiroslavPetrik) [inline tsconfig.json]: ktore _javascript_ subory chceme kompilovat?

Comment (MiroslavPetrik) [reply tsconfig.json]: ak uz ich dovolime, nechces zapnut _checkJs_?

File: tsconfig.json [hunk 7]
```diff
@@ -11,6 +11,7 @@
     "module": "esnext",
     "moduleResolution": "node",
     "noEmit": true,
+    "noUnusedLocals": true,
```
Comment (MiroslavPetrik) [inline tsconfig.json]: mozno включи _noUnusedParameters_ as well?

File: tsconfig.json (hunk 6 above)
Comment (Kamahl19) [reply tsconfig.json]: ak to zmazem, CRA mi to forne na true. Ok, necham false.

**Pull request merged**